### ISSUE TITLE
What is the nature of this error?

### ISSUE THREAD

**Issue by DaTrader** [phoenixframework/phoenix_live_view#1405]
### Environment

* Elixir version (elixir -v): 1.10.3
* Phoenix version (mix deps): 1.5.8
* Phoenix LiveView version (mix deps): 0.15.4
* NodeJS version (node -v): v12.14.1
* NPM version (npm -v): 6.13.4
* Operating system: Linux Mint
* Browsers you attempted to reproduce this bug on (the more the merrier): Firefox, Chrome
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual behavior

I sometimes get this error in my JS console but although its not too difficult to get it, I've never managed to reproduce the exact steps of getting it. It's more like trying to repeat more of the same which comes down to the following:
- have some elements created on the page
- have some destroyed
- shut down the server
- restart the server
- have elements created/destroyed via fast sequence of messages (keeping pressing down a key that collapses/expands a number of items in the appended container list).

Can someone please tell me, from the LiveView js perspective, where it breaks and what may be the cause of the error.

Uncaught TypeError: n is undefined
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:955
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:957
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:867
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:980
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:966
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:953
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:957
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:867
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:980
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:966
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:953
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:957
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:957
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:867
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:855
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2486
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:1093
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2483
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2474
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:1093
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2473
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2555
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2291
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2552
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:1121
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:405
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:826
    decode webpack:///../deps/phoenix/priv/static/phoenix.js?:456
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:816
    onmessage webpack:///../deps/phoenix/priv/static/phoenix.js?:611
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:610
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:692
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:671
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:683
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:670
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:578
    timer webpack:///../deps/phoenix/priv/static/phoenix.js?:1164
    setTimeout handler*value webpack:///../deps/phoenix/priv/static/phoenix.js?:1163
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:697
    onclose webpack:///../deps/phoenix/priv/static/phoenix.js?:613
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:612
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:692
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:671
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:683
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:670
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:578
    timer webpack:///../deps/phoenix/priv/static/phoenix.js?:1164
    setTimeout handler*value webpack:///../deps/phoenix/priv/static/phoenix.js?:1163
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:697
    onclose webpack:///../deps/phoenix/priv/static/phoenix.js?:613
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:612
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:692
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:671
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:683
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:670
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:578
    timer webpack:///../deps/phoenix/priv/static/phoenix.js?:1164
    setTimeout handler*value webpack:///../deps/phoenix/priv/static/phoenix.js?:1163
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:697
    onclose webpack:///../deps/phoenix/priv/static/phoenix.js?:613
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:612
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:692
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:671
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:683
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:670
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:578
    timer webpack:///../deps/phoenix/priv/static/phoenix.js?:1164
    setTimeout handler*value webpack:///../deps/phoenix/priv/static/phoenix.js?:1163
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:697
    onclose webpack:///../deps/phoenix/priv/static/phoenix.js?:613
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:612
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672
app.js line 119 > eval:955:11
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:955
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:957
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:867
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:980
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:966
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:953
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:957
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:867
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:980
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:966
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:953
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:957
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:972
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:957
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:867
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:855
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2486
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:1093
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2483
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2474
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:1093
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2473
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2555
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2291
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:2552
    value webpack:///../deps/phoenix_live_view/priv/static/phoenix_live_view.js?:1121
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:405
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:826
    decode webpack:///../deps/phoenix/priv/static/phoenix.js?:456
    decode self-hosted:1173
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:816
    onmessage webpack:///../deps/phoenix/priv/static/phoenix.js?:611
    (Async: EventHandlerNonNull)
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:610
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:692
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:671
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:683
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:670
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:578
    timer webpack:///../deps/phoenix/priv/static/phoenix.js?:1164
    (Async: setTimeout handler)
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:1163
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:697
    onclose webpack:///../deps/phoenix/priv/static/phoenix.js?:613
    (Async: EventHandlerNonNull)
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:612
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:692
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:671
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:683
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:670
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:578
    timer webpack:///../deps/phoenix/priv/static/phoenix.js?:1164
    (Async: setTimeout handler)
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:1163
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:697
    onclose webpack:///../deps/phoenix/priv/static/phoenix.js?:613
    (Async: EventHandlerNonNull)
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:612
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:692
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:671
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:683
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:670
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:578
    timer webpack:///../deps/phoenix/priv/static/phoenix.js?:1164
    (Async: setTimeout handler)
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:1163
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:697
    onclose webpack:///../deps/phoenix/priv/static/phoenix.js?:613
    (Async: EventHandlerNonNull)
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:612
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:692
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:671
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:683
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:670
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:578
    timer webpack:///../deps/phoenix/priv/static/phoenix.js?:1164
    (Async: setTimeout handler)
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:1163
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:697
    onclose webpack:///../deps/phoenix/priv/static/phoenix.js?:613
    (Async: EventHandlerNonNull)
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:612
    reconnectTimer webpack:///../deps/phoenix/priv/static/phoenix.js?:579
    value webpack:///../deps/phoenix/priv/static/phoenix.js?:672

**Comment by DaTrader** [comment]
Below is a screenshot of the line where it breaks

![image](https://user-images.githubusercontent.com/30151179/114277607-1a92d100-9a2c-11eb-8e36-5a0ef4768220.png)

**Comment by josevalim** [comment]
It is not possible to say without a way to reproduce because line 955 points to a function definition on your tag, which is not possible to be a type error:

https://github.com/phoenixframework/phoenix_live_view/blob/v0.15.4/assets/js/phoenix_live_view.js#L955

**Comment by josevalim** [comment]
Ah, the screenshot helps. I believe there is a report for this or it has been addressed. Let me check.

**Comment by josevalim** [comment]
I believe this is the same as #1144 addressed by #1393.

**Comment by DaTrader** [comment]
Yes, it does look as the same error. Just read the whole conversation there and am as confused as the other guys. Hail to the person who managed to reproduce it though.

**Comment by DaTrader** [comment]
Ah, yes now the other guy with the remarks on conditional rendering reminded me that I can additionally reproduce a very similar error (with a somewhat shorter stack trace - from the line "onmessage webpack:///../deps/phoenix/priv/static/phoenix.js?:611" up to the same line as in the provided stack trace and screenshot.

This other case which I now avoid (and have been asking for the LiveComponent.render/2 for) relates to the following conditional rendering in a component template:
```

<div id="<%= @id %>
  <%= if @delete do %>
     data-delete="true"
  <% else %>
     [ all other attributes including the hook]
  <% end %>>
  <%= unless @delete do %>
    [ the nested elements in the standard template of this component]
  <% end %>
</div>

```

This was one of the things I tried to force my way into having component templates in the same folder, but as I said, the above use to fail over the same or extremely similar error immediately and every single time unlike now when I no longer have such structure in the templates. 

However, it is strange though because I still use some simple conditional rendering (for single attributes) and it works. But I now tend to optimize them into totally static ones with just the value being conditioned e.g. data-something="<%= if @foo do %>bar<% end %>" as then the attribute tag only gets passed once over the wire while conditioning it in full has the entire tag text passed each time the condition is met.

**Comment by josevalim** [comment]
This should be fixed on master by #1393. If it isn't, please provide a reproducible app, thanks!
